bavard_ml_utils/persistence/record_store/dynamodb.py

```diff
@@ -91,24 +93,51 @@ def delete_all(self, *conditions: t.Tuple[str, str, t.Any], **where_equals) -> i
         # TODO: This is an incredibly slow way of doing this. Use indexes and a batch delete instead.
         self.assert_can_edit()
         num_deleted = 0
-        for record in self.get_all(*conditions, **where_equals):
+        # for record in self.get_all(*conditions, **where_equals):
+        for record in self._scan():
```

Comment:
This PR looks great @Alifilsoofi. Do we need to pass the conditions to `self._scan()` here?

Reply:
@epeters3 Nice catch!